scripts/tables_2023.R

```diff
@@ -0,0 +1,247 @@
+  summarise(catch = n()) %>%
+  arrange(site_id, species_code, ef) %>%
+  # ungroup() %>%
+  mutate(catch = case_when(
+    species_code == 'NFC' ~ 0L,
+    T ~ catch),
+    # effort = catch/ef_seconds,
+    id = paste0(local_name, '_', species_code, '_', life_stage)) %>%
+  ungroup() %>%
+  arrange(id)
+
+# join the total number of passes to each event so that we know if it is a higher number than the pass of the catch
+fish_abund_prep2 <- left_join(
+  fish_abund_prep,
+
+  fish_abund_prep %>%
+    group_by(local_name) %>%
+    summarise(pass_total = max(haul_number_pass_number)),
+  by = 'local_name'
+)
+
+# make a dat to indicate if the nfc in the set for each species
+fish_nfc_tag<- fish_abund_prep2 %>%
+  mutate(nfc_pass = case_when(
+    # species_code != 'NFC' &
+    haul_number_pass_number == pass_total ~ F,
+    T ~ T),
+    nfc_pass = case_when(
+      species_code == 'NFC' ~ T,
+      T ~ nfc_pass)
+  ) %>%
+  select(local_name, species_code, life_stage, haul_number_pass_number, pass_total, nfc_pass) %>%
+  arrange(desc(haul_number_pass_number)) %>%
+  # filter(nfc_pass == T) %>%
+  distinct(local_name, species_code, life_stage, .keep_all = T) %>%
+  select(-haul_number_pass_number, -pass_total)
+
+# calculate abundance for each site regardless of whether a nfc_pass occurred.
+fish_abund_prep3 <- left_join(
+  fish_abund_prep2 %>%
+    group_by(local_name, species_code, life_stage) %>%
+    summarise(catch = sum(catch)),
+
+  fish_nfc_tag,
+
+  by = c('local_name', 'species_code', 'life_stage')
+)
+
+
+# add back the size of the sites so we can do a density
+fish_abund <- left_join(
+  fish_abund_prep3,
+
+  hab_fish_collect_info %>%
+    select(local_name,
+           # sampling_method,
+           # haul_number_pass_number,
+           ef_seconds:enclosure) %>%
+    distinct(local_name, ef_length_m, .keep_all = T),
+
+  by = c('local_name')
+) %>%
+  mutate(area_m2 = round(ef_length_m * ef_width_m,1),
+         density_100m2 = round(catch/area_m2 * 100,1)) %>%
+  tidyr::separate(local_name, into = c('site', 'location', 'ef'), remove = F)
+
+# Rename so we can call fpr_table_fish_density() and fpr_plot_fish_box() with updated data
+fish_abund_updated <- fish_abund
+
+
+
+## Needed for fpr_table_fish_site()
+### density results -----------------------------------------------------------
+
+# need to summarize just the sites
+tab_fish_sites_sum <- left_join(
+  fish_abund_prep2 %>%
+    select(local_name, pass_total) %>%
+    distinct(),
+
+
+  hab_fish_collect_info %>%
+    select(local_name,
+           ef_length_m:enclosure) %>%
+    distinct(),
+
+  by = 'local_name'
+) %>%
+  mutate(area_m2 = round(ef_length_m * ef_width_m,1)) %>%
+  select(site = local_name, passes = pass_total, ef_length_m, ef_width_m, area_m2, enclosure)
+
+rm(fish_abund_prep,
+  fish_abund_prep2,
+  fish_abund_prep3,
+  fish_nfc_tag
+)
+
+# Rename so we can call fpr_table_fish_site() with updated data
+tab_fish_sites_sum_updated <- tab_fish_sites_sum
```

Comment:
This is a great approach.

One thing to consider is that we will be back to the same sites again in 2024 so nameing as `_2023` might be more robust of an approach.